packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -655,6 +658,56 @@ class DropdownButton<T> extends StatefulWidget {
   /// {@endtemplate}
   final ValueChanged<T> onChanged;
 
+  /// A builder to generate custom dropdown buttons corresponding to the
+  /// [DropdownMenuItem]s in [items].
+  ///
+  /// When a [DropdownMenuItem] is selected, this builder can be used to create
+  /// custom dropdown buttons. The widget that will be displayed from the list
+  /// corresponds to the [DropdownMenuItem] of the same index in [items].
+  ///
+  /// {@tool snippet --template=stateful_widget_scaffold}
+  ///
+  /// This sample shows a `DropdownButton` with a button with [Text] that
+  /// corresponds to but is unique from [DropdownMenuItem].
+  ///
+  /// ```dart
+  /// final List<String> items = <String>[
```

Comment:
A slightly more realistic example might be to have the selected items just be '1', '2', '3'. Sometimes apps need to keep the dropdown buttons horizontally compact, even though the menu items are not.